pkg/db/interface.go

```diff
@@ -341,6 +343,201 @@ func (d *dbConn) DeleteStudy(id string) error {
 	return err
 }
 
+func (d *dbConn) CreateNAS(in *api.StudyConfig) (string, error) {
+
+	for _, operation := range in.NasConfig.Operations.Operation {
+		if len(operation.OperationType) == 0 && operation.ParameterConfigs != nil {
+			return "", errors.New("Operations must be set properly")
+		}
+	}
+
+	if in.JobId != "" {
+		/* BETTER WAY TO CHECK IF THE OBJECT EXISTS */
+		var temporaryId string
+		err := d.db.QueryRow("SELECT id FROM studies WHERE job_id = ?", in.JobId).Scan(&temporaryId)
+		if err == nil {
+			return "", fmt.Errorf("Study %s in Job %s already exist.", in.Name, in.JobId)
+		}
+		// var err error
+		// var count int
+
+		// for row.Next() {
+		// 	err = row.Scan(&count)
+		// 	if err != nil || count == 0 {
+		// 		return "", fmt.Errorf("Study %s in Job %s already exist.", in.Name, in.JobId)
+		// 	}
+		// }
+	}
+
+	nasConfig, err := (&jsonpb.Marshaler{}).MarshalToString(in.NasConfig)
+	if err != nil {
+		log.Fatalf("Error marshaling nasConfig: %v", err)
+	}
+
+	tags := make([]string, len(in.Tags))
+	for i, elem := range in.Tags {
+		tags[i], err = (&jsonpb.Marshaler{}).MarshalToString(elem)
+		if err != nil {
+			log.Printf("Error marshalling %v: %v", elem, err)
+			continue
+		}
+	}
+
+	/* WE PROBABLY DON'T NEED METRICS AND ALSO THIS LOGIC IS KIND OF CONFUSING */
+	var isin bool = false
+	for _, m := range in.Metrics {
+		if m == in.ObjectiveValueName {
+			isin = true
+		}
+	}
+
+	if !isin {
+		in.Metrics = append(in.Metrics, in.ObjectiveValueName)
+	}
+
+	var studyID string
+	var trials string
+	var configs string
+
+	i := 3
+	for true {
```

Comment:
It looks the same process with CreateStudy. Please merge them and split only different process.